File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.swift
func solution(_ num:Int) -> Int {
    var tmpNum = num
    var ans = 0
    
    while tmpNum > 1 {
        if ans == 500 {
            return -1
        }
        if tmpNum % 2 == 0 {
            tmpNum /= 2
            ans += 1
        } else {
            tmpNum = tmpNum * 3 + 1
            ans += 1
        }
    }
    return ans
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        long n = num;
        
        while(n != 1 && answer < 500) {
            if (n % 2 == 0){
                n /= 2;
            } else if (n % 2 == 1) {
                n = n * 3 + 1;
            }
            answer++;
        
        }
        
        return n == 1 ? answer : -1;
    }
}